### PR TITLE
Fix permission change endpoint & add polling

### PR DESCRIPTION
**Pull request by HappyDOGE**

### CODE REVIEW

File: src/lib/Routes/Users.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ function usersAPI(db: JsonDB.JsonDB) {
 			if (req.body.permissionLevel != undefined &&
 				(requestUser.permissionLevel > user.permissionLevel || requestUser.permissionLevel == PermissionLevel.Owner) && // can't change the permission level of a user with the same permission level as you, unless you're owner
 				(requestUser.permissionLevel > req.body.permissionLevel || requestUser.permissionLevel == PermissionLevel.Owner) && // can't change to your own permission level, unless you're owner
-				(requestUser.authProvider != user.authProvider && requestUser.authId != user.authId) && // can't change your own permission level
+				(requestUser.authProvider == user.authProvider && requestUser.authId != user.authId) && // can't change your own permission level
```
Comment (april83c) [inline src/lib/Routes/Users.ts]: yeah this was wrong but wouldn't this make it so you can only change the permissionLevel of users that use the same authProvider as you

File: src/lib/Routes/Notify.ts [hunk 2]
```diff
@@ -44,19 +52,17 @@ function notifyAPI(db: JsonDB.JsonDB, listeners: Array<wsWithHeartbeat>) {
 			});
 		}
 	
-		let notificationJson = JSON.stringify({
+		let notificationObject = {
```
Comment (april83c) [inline src/lib/Routes/Notify.ts]: why isn't type specified here

**Review by april83c — APPROVED** [review]
LGTM, if this has been tested and works (please confirm) i'm ready to merge

File: src/lib/Routes/Users.ts (hunk 1 above)
Comment (HappyDOGE) [reply src/lib/Routes/Users.ts]: yea I fixed it in a followup commit sry

File: src/lib/Routes/Notify.ts (hunk 2 above)
Comment (HappyDOGE) [reply src/lib/Routes/Notify.ts]: I fixed it in my fork but the new commit doesnt show up in the PR 🤔 
https://github.com/HappyDOGE/broadcaster/commit/539391fb69ed4db1a46a230fc308fa97581eaa60

**Comment by HappyDOGE** [comment]
ok, it's here now. I tested this with https://github.com/osuplace/templateManager/pull/39 and it works

File: src/lib/Routes/Listen.ts [hunk 3]
```diff
@@ -10,14 +10,49 @@
...
+	let notificationsPollListeners = 0;
+
+	function sendNotification(notificationObject: NotificationObject) {
+		let notificationJson = JSON.stringify(notificationObject);
+		for (const ws of listeners) {
+			ws.send(notificationJson, (err: any) => { if (err) console.error('Failed to send notification: ' + err); });
+		}
+
+		notificationObject.a = new Date();
+		notificationObject.i = (+notificationObject.a).toString();
```
Comment (april83c) [inline src/lib/Routes/Listen.ts:37]: this property seems redundant since its just a but as a number but whatever

File: src/lib/Routes/Listen.ts [hunk 4]
```diff
@@ -61,7 +96,12 @@ function listenAPI() {
 		listeners.push(ws);
 	});
 
-	return { listenAPIRouter, listeners };
+	listenAPIRouter.get("/listen-poll", (req, res) => {
+		notificationsPollRequests++;
+		res.send(JSON.stringify(notificationsBacklog));
```
Comment (april83c) [inline src/lib/Routes/Listen.ts:101]: this shouldve been res.json by the way

i didnt catch it oops

**Pull request merged**